pkg/controller/signup_check.go

```diff
@@ -0,0 +1,66 @@
+package controller
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/codeready-toolchain/registration-service/pkg/configuration"
+	"github.com/codeready-toolchain/registration-service/pkg/errors"
+	"github.com/gin-gonic/gin"
+)
+
+// SignupCheck implements the SignupCheck endpoint.
+type SignupCheck struct {
+	config               *configuration.Registry
+	logger               *log.Logger
+	testRequestTimestamp int64
+}
+
+// SignupCheckPayload payload
+type SignupCheckPayload struct {
+	ProvisioningDone bool `json:"provisioning_done"`
+}
+
+// NewSignupCheck returns a new SignupCheck instance.
+func NewSignupCheck(logger *log.Logger, config *configuration.Registry) *SignupCheck {
+	return &SignupCheck{
+		logger: logger,
+		config: config,
+	}
+}
+
+// getSignupCheckInfo returns the SignupCheck info.
+func (hc *SignupCheck) getSignupCheckInfo() *SignupCheckPayload {
+	return &SignupCheckPayload{
+		ProvisioningDone: true,
+	}
+}
+
+// GetHandler returns a default heath check result.
+func (hc *SignupCheck) GetHandler(ctx *gin.Context) {
+	// Default handler for system SignupCheck
+	ctx.Writer.Header().Set("Content-Type", "application/json")
+	SignupCheckInfo := hc.getSignupCheckInfo()
+	if hc.config.IsTestingMode() {
+		// testing mode, wait 5s from the first request to flag state complete.
+		if hc.testRequestTimestamp == 0 {
```

Comment:
I'm not a fan of adding too many test specifics to the production code. Let's minimize it as much as possible. It makes code not very clear and also it's error prone. There are many ways to avoid it here and still be able to test it. You could mock `getSignupCheckInfo()` in the test for example.

Reply:
Made that a bit more ligthweight in 7f970f2. I guess this will change anyway in the later PR where we integrate everything together and actually implementing the reporting.